File: backend/src/core/entities/task_submission.py
from pydantic import BaseModel, ConfigDict, Field
from typing import Optional
from decimal import Decimal
from datetime import datetime

class TaskSubmission(BaseModel):
    """Entidade representando o envio de exercício por um aluno."""
    model_config = ConfigDict(frozen=True, extra="forbid")

    id: Optional[str] = Field(None, description="UUID único do envio de tarefa")
    task_id: str = Field(..., description="UUID da tarefa associada")
    user_id: str = Field(..., description="UUID do aluno (JWT extraído/BOLA-safe)")
    selected_option: Optional[str] = Field(None, min_length=1, max_length=10, description="Opção marcada (múltipla escolha)")
    text_answer: Optional[str] = Field(None, description="Texto escrito pelo aluno (questões discursivas)")
    score: Optional[Decimal] = Field(None, description="Nota de 0.00 a 10.00")
    status: str = Field("pending_auto", description="Status da submissão (pending_auto, pending_review, graded)")
    teacher_feedback: Optional[str] = Field(None, description="Feedback textual do instrutor")
    graded_by: Optional[str] = Field(None, description="UUID do instrutor que corrigiu")
    submitted_at: Optional[datetime] = None
    graded_at: Optional[datetime] = None

File: lawrence/lib/features/tasks/domain/entities/task_submission.dart
class TaskSubmission {
  const TaskSubmission({
    required this.id,
    required this.taskId,
    required this.status,
    this.selectedOption,
    this.textAnswer,
    this.score,
    this.teacherFeedback,
    this.submittedAt,
    this.idempotencyKey,
    this.correctOption,
    this.isCorrect,
  });

  final String id;
  final String taskId;
  final String status;
  final String? selectedOption;
  final String? textAnswer;
  final double? score;
  final String? teacherFeedback;
  final DateTime? submittedAt;
  final String? idempotencyKey;
  final String? correctOption;
  final bool? isCorrect;

  factory TaskSubmission.fromJson(Map<String, dynamic> json) => TaskSubmission(
    id: json['id'] as String? ?? '',
    taskId: json['task_id'] as String? ?? '',
    status: json['status'] as String? ?? 'pending_review',
    selectedOption: json['selected_option'] as String?,
    textAnswer: json['text_answer'] as String?,
    score: (json['score'] as num?)?.toDouble(),
    teacherFeedback: json['teacher_feedback'] as String?,
    submittedAt: json['submitted_at'] == null
        ? null
        : DateTime.tryParse(json['submitted_at'].toString()),
    idempotencyKey: json['idempotency_key'] as String?,
    correctOption: json['correct_option'] as String?,
    isCorrect: json['is_correct'] as bool?,
  );
}
